src/blockchain/bitcoin_rpc.rs

```diff
@@ -0,0 +1,392 @@
+use std::collections::HashSet;
+
+#[allow(unused_imports)]
+use log::{debug, error, info, trace};
+
+use bitcoin::{
+    util::bip32::{ChildNumber, DerivationPath},
+    Address, Network, OutPoint, Transaction, Txid,
+};
+
+use bitcoincore_rpc::{
+    json::{
+        ImportMultiOptions, ImportMultiRequest, ImportMultiRequestScriptPubkey,
+        ImportMultiRescanSince,
+    },
+    Client, RpcApi,
+};
+
+use super::*;
+use crate::database::{BatchDatabase, BatchOperations, DatabaseUtils};
+use crate::error::Error;
+use crate::types::{ScriptType, TransactionDetails, UTXO};
+
+pub struct BitcoinRpcBlockchain(Option<Client>);
+
+impl BitcoinRpcBlockchain {
+    // FIXME: why can I just say BitcoinRpcBlockchain(Some(client))?
+    pub fn new(client: Option<Client>) -> Self {
+        Self(client)
+    }
+    async fn index_is_synced<D: BatchDatabase + DatabaseUtils>(
+        &mut self,
+        index: u32,
+        database: &mut D,
+    ) -> Result<bool, Error> {
+        let networks = vec![Network::Bitcoin, Network::Testnet, Network::Regtest];
+        let path = DerivationPath::from(vec![ChildNumber::Normal { index }]);
+        let last_script = database
+            .get_script_pubkey_from_path(ScriptType::External, &path)?
+            .unwrap();
+        let synced = networks
+            .iter()
+            .map(|network| {
+                let address = Address::from_script(&last_script, *network).unwrap();
+                let response = self.0.as_mut().unwrap().get_address_info(&address);
+                if let Ok(result) = response {
+                    if let Some(is_watchonly) = result.is_watchonly {
+                        return is_watchonly;
+                    }
+                }
+                false
+            })
+            .any(|x| x);
```

Comment:
I'm trying to determine if Bitcoin Core knows about this script.

The only method of determining this that I know of is `getaddressinfo`, which requires we actually turn the script into an address. Since this module has not notion of "network", I try each one.